### PR TITLE
Replace memoryone with memory_depth, various PEP8, a few integer division corrections

### LINKED ISSUE #223: Memory-depth
The memoryone label could be more informative, indicating which strategies are memory-2, memory-3, or memory-infinity. This occur to me while adding HARD_TFT and a few others.

If no one objects I can easily go through and update the bool to an integer as appropriate.

### PR DESCRIPTION
**Pull request by marcharper**
Fixes #223, strategies now have their memory_depth indicated as follows:
* 0 if the last round is irrelevant (ALLC, ALLD, and Random)
* 1 if the strategy is memory one
* n if the strategy uses the last n rounds (e.g. 2 for TitFor2Tats, 3 for HARD_TFT, various for GoByMajority)
* infinity aka float('inf') if the entire history is used or there is an internal bit of memory that can persist indefinitely.
* For Gellar, since it looks into the future, I set its memory_depth to -1.

While I was going through all the strategy files I made a few various updates e.g. for PEP8, and added in float(...) in a few places where it looked like there was unintentional integer division (e.g. in forgiver). Lastly, Darwin had a game matrix hard-coded, so I updated it to pull from the Game class.

(Edited to link PR with issue in waffle)

### CODE REVIEW

**Comment by drvinceknight** [comment]
> For Gellar, since it looks into the future, I set its memory_depth to -1.

Neat!

> While I was going through all the strategy files I made a few various updates e.g. for PEP8, and added in float(...) in a few places where it looked like there was unintentional integer division (e.g. in forgiver). Lastly, Darwin had a game matrix hard-coded, so I updated it to pull from the Game class.

Sweet! Great bunch of fixes.

Looks like travis failed in a couple of places:

```
Traceback (most recent call last):
  File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/tests/unit/test_punisher.py", line 50, in test_strategy
    for turn in range(P1.mem_length-1):
TypeError: range() integer end argument expected, got float.
```

**Comment by marcharper** [comment]
Looks like the integer division was intentional, and I fixed the test (just based on random outcomes).

**Comment by drvinceknight** [comment]
Cool, will check through this in the am. (tired... :zzz: :sleeping: )

File: axelrod/strategies/oncebitten.py [hunk 1]
```diff
@@ -52,11 +52,11 @@ def reset(self):
 
 class FoolMeOnce(Player):
     """
-    Forgives one D then retaliates forever on a second D.
+    Forgives one D then retaliates forever on a second D. Also known as GRIM.
```
Comment (drvinceknight) [inline axelrod/strategies/oncebitten.py]: This is not correct. Grudger is also known as `GRIM`.

File: axelrod/strategies/oncebitten.py [hunk 2]
```diff
@@ -87,9 +87,9 @@ class ForgetfulFoolMeOnce(Player):
     """
 
     name = 'Forgetful Fool Me Once'
-    memoryone = False  # Long memory
+    memory_depth = float('inf')  # Long memory
 
-    def __init__(self, forget_probability=0.1):
+    def __init__(self, forget_probability = 0.05):
```
Comment (drvinceknight) [inline axelrod/strategies/oncebitten.py]: Why this change?

File: axelrod/strategies/oncebitten.py (hunk 1 above)
Comment (marcharper) [inline axelrod/strategies/oncebitten.py]: Fixed

File: axelrod/strategies/oncebitten.py (hunk 2 above)
Comment (marcharper) [inline axelrod/strategies/oncebitten.py]: 0.1 was arbitrary -- 0.05 seemed like it might be better (more like typical GTFT values?). As far as I know I just made this strategy up, so there's no standard choice.

Comment (drvinceknight) [inline axelrod/strategies/oncebitten.py]: Cool.
